enemy/enemy_spawn.gd

```diff
@@ -1,16 +1,54 @@
 extends Node2D
+class_name EnemySpawn
 
 const FlyingEye = preload("res://enemy/flying_eye/flying_eye.tscn")
 const Goblin = preload("res://enemy/goblin/goblin.tscn")
 const Skeleton = preload("res://enemy/skeleton/skeleton.tscn")
 const Mushroom = preload("res://enemy/mushroom/mushroom.tscn")
 
-const enemy_variants: Array[PackedScene] = [FlyingEye, Goblin, Skeleton, Mushroom]
+# Dictionary that maps diffuclty score to a set of enemies
+const enemies: Dictionary = {
+	1: [FlyingEye],
+	2: [Goblin],
+	4: [Skeleton],
+	6: [Mushroom]
+}
```

Comment:
This doesn't need to be this way, but found this to be the easiest path for now, since they are not classes that expose a score or anything like that, and they don't self-register either.

Reply:
i was wondering how we might solve when which enemies were introduced